### PR TITLE
EDSC-3463: Adds subscription editing

### PR DESCRIPTION
**Pull request by trevorlang**
# Overview

### What is the feature?

Adds a modal to allow editing of subscription items

### What is the Solution?

Implement a modal

### What areas of the application does this impact?

Granule and Collection subscriptions lists

# Testing

### Reproduction steps

- **Environment for testing:** SIT
- **Collection to test with:** ANY

1. Add a subscription
2. Edit only the name
3. Confirm the name is changed

1. Edit a subscription
2. Change parameters
3. Confirm the parameters are changed

1. Confirm validation prevents editing of subscriptions with matching queries (the edit button is disabled)

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/containers/EditSubscriptionModalContainer/__tests__/EditSubscriptionModalContainer.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  const enzymeWrapper = shallow(<EditSubscriptionModalContainer {...props} />)
+
+  return {
+    enzymeWrapper,
+    props
+  }
+}
+
+describe('mapDispatchToProps', () => {
```
Comment (macrouch) [inline static/src/js/containers/EditSubscriptionModalContainer/__tests__/EditSubscriptionModalContainer.test.js:29]: Missing a test here for `onUpdateSubscription`

File: static/src/js/components/EditSubscriptionModal/EditSubscriptionModal.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
+/* eslint-disable max-len */
```
Comment (macrouch) [inline static/src/js/components/EditSubscriptionModal/EditSubscriptionModal.js:1]: Move to disable the specific line, but is it necessary?

File: static/src/js/actions/__tests__/subscriptions.test.js [hunk 3]
```diff
@@ -897,20 +898,16 @@ describe('updateSubscription', () => {
...
-          query: 'browse_only=true&temporal=2020-01-01T00:00:00.000Z,2020-01-31T23:59:59.999Z'
-        }
-      })
 
       expect(addToastMock.mock.calls.length).toBe(1)
       expect(addToastMock.mock.calls[0][0]).toEqual('Subscription updated')
       expect(addToastMock.mock.calls[0][1].appearance).toEqual('success')
       expect(addToastMock.mock.calls[0][1].autoDismiss).toEqual(true)
+
+      console.log('getGranuleSubscriptionsMock.mock.calls', getGranuleSubscriptionsMock.mock.calls)
```
Comment (macrouch) [inline static/src/js/actions/__tests__/subscriptions.test.js:907]: remove

File: static/src/js/components/EditSubscriptionModal/EditSubscriptionModal.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      subscriptionConceptId: subscriptionConceptIdFromProps
+    } = props
+
+    const { subscription: subscriptionFromState = {} } = state
+    const { conceptId: subscriptionConceptIdFromState } = subscriptionFromState
+
+    // Check to see if a new subscription has been loaded
+    if (subscriptionConceptIdFromState === subscriptionConceptIdFromProps) return null
+
+    // TODO: is this the best way to handle this?
```
Comment (trevorlang) [inline static/src/js/components/EditSubscriptionModal/EditSubscriptionModal.js:40]: use subscriptionType

File: static/src/js/components/EditSubscriptionModal/EditSubscriptionModal.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    this.setState({
+      shouldUpdateQuery: checked
+    })
+  }
+
+  onModalClose() {
+    const { onToggleEditSubscriptionModal } = this.props
+    onToggleEditSubscriptionModal({
+      isOpen: false,
+      subscriptionConceptId: ''
```
Comment (trevorlang) [inline static/src/js/components/EditSubscriptionModal/EditSubscriptionModal.js:105]: Set type here too

**Review by macrouch — APPROVED** [review]

**Review by DanielJDufour — APPROVED** [review]

**Pull request merged**